api/models/zones.go

```diff
@@ -0,0 +1,70 @@
+package models
+
+import (
+	"context"
+	"fmt"
+	"math"
+
+	"github.com/PedroChaparro/loomies-backend/configuration"
+	"github.com/PedroChaparro/loomies-backend/interfaces"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var ZonesCollection *mongo.Collection = configuration.ConnectToMongoCollection("zones")
+
+func GetNearGyms(currentLatitude float64, currentLongitude float64) (p []interfaces.Gym, e error) {
+
+	// initial zones calculations
+	const initialLatitude = 6.9595
+	const initialLongitude = -73.1696
+	const sizeMinZone = 0.0035
+
+	initialX := math.Floor((currentLongitude - initialLongitude) / sizeMinZone)
+	initialY := math.Floor((currentLatitude - (initialLatitude)) / sizeMinZone)
+
+	var mZonesCoord []string
+	mZonesCoord = append(mZonesCoord, fmt.Sprintf("%v,%v", initialX-1, initialY+1)) // Box Top Left
+	mZonesCoord = append(mZonesCoord, fmt.Sprintf("%v,%v", initialX, initialY+1))   // Box Top - North
+	mZonesCoord = append(mZonesCoord, fmt.Sprintf("%v,%v", initialX+1, initialY+1)) // Box Top Right
+	mZonesCoord = append(mZonesCoord, fmt.Sprintf("%v,%v", initialX-1, initialY))   // Box Left
+	mZonesCoord = append(mZonesCoord, fmt.Sprintf("%v,%v", initialX, initialY))     // current zone box
+	mZonesCoord = append(mZonesCoord, fmt.Sprintf("%v,%v", initialX+1, initialY))   // Box Right
+	mZonesCoord = append(mZonesCoord, fmt.Sprintf("%v,%v", initialX-1, initialY-1)) // Box Bottom Left
+	mZonesCoord = append(mZonesCoord, fmt.Sprintf("%v,%v", initialX, initialY-1))   // Box Bottom - South
+	mZonesCoord = append(mZonesCoord, fmt.Sprintf("%v,%v", initialX+1, initialY-1)) // Box Bottom Right
```

Comment:
For some reason I always get only 8 zones instead of 9. Try these coordinates: 

```json
"latitude": 7.03823,
"longitude": -73.07137, 
```

Reply:
@SilviaPabon, remove this validation: 

```go
if !cursor.Next(context.Background()) {
    return []interfaces.Gym{}, err
}
```

It is calling the cursor.Next() property, so, always skip the first gym. It's safe to remove it because you're validating the final array length in the controller file.

Reply:
I got it! I update a condition yesterday to get and empty array when there is no gyms but, this is affecting when there are gyms but is the last one:
``` 
if !cursor.Next(context.Background()) {
		return []interfaces.Gym{}, err
	} 

	for cursor.Next(context.Background()) {

		var surroundZones interfaces.ZoneWithGyms
		cursor.Decode(&surroundZones)

		fmt.Println(&surroundZones)
		// some zones don't have gyms
		if len(surroundZones.Gyms) != 0 {
			gyms = append(gyms, surroundZones.Gyms[0])
		}
	}
```
Thanks a lot for this!

Reply:
> @SilviaPabon, remove this validation:
> 
> ```go
> if !cursor.Next(context.Background()) {
>     return []interfaces.Gym{}, err
> }
> ```
> 
> It is calling the cursor.Next() property, so, it's always skipping the first gym. It's safe to remove it because you're validating the final array length in the controller file.

Caramba! I didn't see this message until now! Thanks!